### ISSUE TITLE
[Bug] - New imports are added before folded ones

### ISSUE THREAD

**Issue by mamilic** [redhat-developer/vscode-java#3821]
When I autoimport the suggested claasname, prior to enabling the fold imports, it will add new import before folded ones.
![image](https://github.com/user-attachments/assets/c41553a0-b348-4517-b2c0-3903a026346d)

##### Environment
- Operating System: Windows 11
- JDK version: 21
- Visual Studio Code version: 1.94.2
- Java extension version: 1.35.1

Steps:

1. Open [demo.zip](https://github.com/user-attachments/files/17411567/demo.zip)
2. Navigate to DemoApplication.java and try to add new import by typing claas name instead of ArrayList by autocompleting the classname.
3. It will add new import before the folded ones

**Comment by rgrunber** [comment]
The import order is determined by https://github.com/redhat-developer/vscode-java/blob/cc32926776bbae1c95c7d51ed982604b3c07c249/package.json#L1120-L1133

So the imports starting with "java" will be placed above those starting with "org", in addition to whitespace to indicate they are a separate grouping. You could remove the "org" imports and add them back in to see they are placed below.

**Comment by rgrunber** [comment]
Yup, you're right. The language server is correctly defining the import fold, but I'm still seeing the previous folding range in the UI. You can even see it in your screenshot (the 2 folds). The only way I'm able to get things folded again is by `Toggle Import Fold` (editor.toggleImportFold) from the command palette, or by using a completely fresh vscode instance :stuck_out_tongue_winking_eye:  .

Maybe the folding history is saved for each editor and it conflicts with the ability to update the fold automatically. @jdneo , can you reproduce this ? If toggling the import fold manually can fix this, I would think it's a bug in the automated logic in VS Code for `editor.foldingImportsByDefault`. Thoughts ?

**Comment by mamilic** [comment]
@rgrunber , thank you for detail explanation. There is issue already opened against vscode, https://github.com/microsoft/vscode/issues/192775. However I wasn't sure whether this issue was related to this repo.
